### PR TITLE
Break out the IsFrozen flag to be an aspect of CompilationTrackerState. (first commit)

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
+using System.Resources;
```
Comment (ToddGrun) [inline src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs:11]: Was this required?

File: src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.CompilationTrackerState.cs [hunk 2]
```diff
@@ -120,15 +145,20 @@ private sealed class InProgressState : WithCompilationTrackerState
...
                 public InProgressState(
+                    bool isFrozen,
                     Compilation compilationWithoutGeneratedDocuments,
                     CompilationTrackerGeneratorInfo generatorInfo,
                     Compilation? compilationWithGeneratedDocuments,
                     ImmutableList<(ProjectState state, CompilationAndGeneratorDriverTranslationAction action)> intermediateProjects)
-                    : base(compilationWithoutGeneratedDocuments,
-                           generatorInfo.WithDocumentsAreFinal(false)) // since we have a set of transformations to make, we'll always have to run generators again
+                    : base(isFrozen,
+                           compilationWithoutGeneratedDocuments, generatorInfo)
```
Comment (ToddGrun) [inline src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.CompilationTrackerState.cs:154]: >generatorInfo [](http://example.com/codeflow?start=65&length=13)

nit: newline

Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.CompilationTrackerState.cs:154]:
```suggestion
                           compilationWithoutGeneratedDocuments,
                           generatorInfo)
```

File: src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs]: no clue.  removed :)

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski review this PR if you want something smaller.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**